build_tools/cmake/iree_copts.cmake

```diff
@@ -212,6 +212,7 @@ iree_select_compiler_opts(IREE_DEFAULT_COPTS
     "-Wno-comment"
     "-Wno-format-zero-length"
     "-Wno-uninitialized"
+    "-Wno-overloaded-virtual"  # gcc complains more than clang, see #15631
```

Comment:
Argh, found another warning/error:
```
[build] /usr/local/google/home/scotttodd/code/iree-build/_deps/spirv-tools-src/source/opt/trim_capabilities_pass.cpp:146:17: error: ‘*(unsigned int*)((char*)&<unnamed> + offsetof(std::optional<spv::Capability>,std::optional<spv::Capability>::<unnamed>.std::_Optional_base<spv::Capability, true, true>::<unnamed>))’ may be used uninitialized [-Werror=maybe-uninitialized]
[build]   146 |     return std::nullopt;
[build]       |                 ^~~~~~~
```

I wonder if the compiler plugin changes affected which flags we apply to third party code. Probably need to look at `iree_cc_library`, `external_cc_library` (which has no uses right now?), etc.?

Reply:
I got the following error when testing locally...
cc1: error: command-line option ‘-Wno-overloaded-virtual=0’ is valid for C++/ObjC++ but not for C [-Werror]

Reply:
Yeah, it sounds to me that building the 3rd-party libs with the iree source flags is going to create warning/errors. The issue is that they are having different warning level settings. I think...

Reply:
> I got the following error when testing locally... cc1: error: command-line option ‘-Wno-overloaded-virtual=0’ is valid for C++/ObjC++ but not for C [-Werror]

Whoops, thanks. I changed this to only add the flag for C++.